libraries/AP_Baro/AP_Baro_BMP280.cpp

```diff
@@ -126,24 +145,37 @@ bool AP_Baro_BMP280::_init()
 void AP_Baro_BMP280::_timer(void)
 {
     uint8_t buf[6];
+    uint8_t sts;
+
+    // check status first
+    if(!_dev->read_registers(BMP280_REG_STATUS, &sts, sizeof(sts))) return;
+    if (sts & ((1<<3) | (1<<0)) ) { // if conversion is on or NVM copy in progress
+        return;
+    }
+    
+    // conversion is done, can read data
+    if(!_dev->read_registers(BMP280_REG_DATA, buf, sizeof(buf))) return;
 
-    _dev->read_registers(BMP280_REG_DATA, buf, sizeof(buf));
+    // start new measure
+    _dev->write_register((BMP280_REG_CTRL_MEAS & _mask), (BMP280_OVERSAMPLING_T << 5) | (BMP280_OVERSAMPLING_P << 2) | BMP280_MODE);
+
+    _dev->get_semaphore()->give();  // give bus semaprore ASAP, before calculations
 
     _update_temperature((buf[3] << 12) | (buf[4] << 4) | (buf[5] >> 4));
-    _update_pressure((buf[0] << 12) | (buf[1] << 4) | (buf[2] >> 4));
+    _update_pressure(   (buf[0] << 12) | (buf[1] << 4) | (buf[2] >> 4));
+
+    return;
 }
 
 // transfer data to the frontend
 void AP_Baro_BMP280::update(void)
 {
     if (_sem->take_nonblocking()) {
-        if (!_has_sample) {
-            _sem->give();
-            return;
+        if (_has_sample) {
+            float pressure = _pressure_filter.getf();
```

Comment:
_pressure_filter is not updated. So driver didnt work at all at current from!

Reply:
Oops... it was lost preparing PR to upstream. Fixed now

Reply:
Thank you